tests/unit/AddressChangeTest.php

```diff
@@ -32,4 +43,53 @@ public function testWhenAddressIdentifierIsUpdated_dataIsProperlyAssigned() {
 		$this->assertSame( $initialIdentifier, $addressChange->getPreviousIdentifier() );
 		$this->assertNotSame( $initialIdentifier, $addressChange->getCurrentIdentifier() );
 	}
+
+	public function testWhenMembershipApplicationIsCreated_addressChangeUuidIsStored(): void {
+		$application = new MembershipApplication();
+
+		$oldId = $application->getAddressChange()->getCurrentIdentifier();
+		$this->entityManager->persist( $application );
+		$this->entityManager->flush();
+
+		/** @var MembershipApplication $persistedApplication */
+		$persistedApplication = $this->entityManager->find( MembershipApplication::class, 1 );
+		$this->assertSame(
+			$oldId,
+			$persistedApplication->getAddressChange()->getCurrentIdentifier()
+		);
+	}
+
+	public function testWhenDonationIsCreated_addressChangeUuidIsStored(): void {
+		$donation = new Donation();
```

Comment:
I'd not put code binding to Donation and to MembershipApplication together. That definitely goes against making the persistence of each BC private to that BC. The idea being to move the Doctrine Donation code into the Donation BCs data access layer and the Doctrine Application into the Membership BCs data access layer, and then deleting this library. ... Well now address also needs to be put somewhere, but that is not as important as making the bulk of the core code private to the correct data access layer.